src/typings/types.ts

```diff
@@ -0,0 +1,132 @@
+import RenderTargetBase from "renderers/RenderTargetBase";
+import { HanziWriterRendererConstructor } from "../renderers/HanziWriterRendererBase";
+
+export type CharacterJson = {
+  strokes: string[];
+  medians: number[][][];
+  radStrokes?: number[];
+};
+
+export type CharDataLoaderFn = (
+  char: string,
+  onLoad: (data: CharacterJson) => void,
+  onError: (err?: Error | string) => void,
+) => Promise<CharacterJson> | CharacterJson | void;
+
+export type Point = { x: number; y: number };
+
+export type ColorObject = { r: number; g: number; b: number; a: number };
+
+export type ColorOptions = {
+  /** (Hex string, Default: "#555"). The color to draw each stroke. */
+  strokeColor: string | null;
+  /** (Hex string, Default: null). The color to draw the radical in the stroke, if radical data is present. Radicals will be drawn the same color as other strokes if this is not set. */
+  radicalColor: string | null;
+  /** (Hex string, Default: "#AAF"). The color to use for highlighting in quizzes. */
+  highlightColor: string | null;
+  /** (Hex string, Default: "#DDD"). The color of the character outline.  */
+  outlineColor: string | null;
+  /** (Hex string, Default: "#333"). The color of the lines drawn by users during quizzing. */
+  drawingColor: string | null;
+  /** (Hex string, Default: null). The color to use when highlighting the character on complete in quizzes. If not set, `highlightColor` will be used instead. Only relevant if `highlightOnComplete` is `true`. */
+  highlightCompleteColor: string | null;
+};
+
+export type OnCompleteFunction = (res: { canceled: boolean }) => void;
+
+/** Creates a render target (e.g. svg, canvas) */
+export type RenderTargetInitFunction<
+  TElement extends HTMLElement | SVGElement | SVGSVGElement | HTMLCanvasElement
+> = (
+  elmOrId: string | TElement,
+  width?: string | number | null,
+  height?: string | number | null,
+) => RenderTargetBase<TElement>;
+
+export type StrokeData = {
+  character: string;
+  drawnPath: {
+    pathString: string;
+    points: Point[];
+  };
+  strokeNum: number;
+  mistakesOnStroke: number;
+  totalMistakes: number;
+  strokesRemaining: number;
+};
+
+export type QuizOptions = {
```

Comment:
nit: for options like this, I'd prefer they be defined at the top of the file that this corresponds to, in this case `Quiz.ts`